src/event-cache/__tests__/EventCache.test.ts

```diff
@@ -192,6 +483,41 @@ describe('EventCache tests', () => {
         expect(eventCache.hasEvents()).toBeFalsy();
     });
 
+    test('when candidate cache size reached, recordEvent drops the newest candidate', async () => {
+        // Init
+        const EVENT1_SCHEMA = 'com.amazon.rum.event1';
+        const EVENT2_SCHEMA = 'com.amazon.rum.event2';
+        const BATCH_LIMIT = 20;
+        const CANDIDATES_LIMIT = 1;
+        const eventCache: EventCache = Utils.createEventCache({
+            ...DEFAULT_CONFIG,
+            ...{
+                batchLimit: BATCH_LIMIT,
+                candidatesCacheSize: CANDIDATES_LIMIT
+            }
+        });
+        const expectedEvents: RumEvent[] = [
+            {
+                id: expect.stringMatching(/[0-9a-f\-]+/),
+                timestamp: new Date(),
+                type: EVENT1_SCHEMA,
+                metadata: `{"version":"1.0.0","aws:client":"${INSTALL_MODULE}","aws:clientVersion":"${WEB_CLIENT_VERSION}"}`,
+                details: '{}'
+            }
+        ];
+
+        // Run
+        eventCache.recordEvent(EVENT1_SCHEMA, {});
+        eventCache.recordEvent(EVENT2_SCHEMA, {});
+        eventCache.recordEvent(EVENT2_SCHEMA, {}, { replaceFirstMatch: true });
+
+        // Assert
+        expect(eventCache.getEventBatch(true)).toEqual(
+            expect.arrayContaining(expectedEvents)
```

Comment:
question: should we also be checking for event with type `EVENT2_SCHEMA`?

Reply:
+1 can we do this check here

Reply:
Updated

Reply:
Rewrote the tests in rev2